app/services/handlers/user/get-user-handler.js

```diff
@@ -0,0 +1,36 @@
+// app/services/handlers/user/get-user-handler.js
+import Service, { inject as service } from '@ember/service';
+
+export default class HandlersUserGetUserHandler extends Service {
+  @service store;
+
+  async request(context, next) {
+    const response = await next(context.request);
+
+    // Handling the GET response and normalizing data for the store
+
+    // Check if the response contains an array of users - which it should
+    if (Array.isArray(response.content.users)) {
+      response.content.users.forEach((user) => {
+        let existingUser = this.store.peekRecord('user', String(user.id));
+        if (!existingUser) {
+          this.store.createRecord('user', {
```

Comment:
At this point, I think we want to "push" some data to the store instead of "create". Create is normally used when a user fills a form and adds a new resource.

Reply:
Thanks Piotr - changing to the push approach like this:

```
if (Array.isArray(response.content.users)) {
      response.content.users.forEach((user) => {
        let existingUser = this.store.peekRecord('user', String(user.id));

        // using push - so the store will expect the correct format - setting up userRecord to be the correct format
        const userRecord = {
          data: {
            id: String(user.id),
            type: 'user',
            attributes: {
              firstName: user.first_name,
              lastName: user.last_name,
              description: user.description,
            }
          }
        };

        // and if exists or not we update the store
        if (!existingUser) {
          this.store.push(userRecord);
        } else {
          existingUser.setProperties({
            firstName: user.first_name,
            lastName: user.last_name,
            description: user.description,
          });
        }
      });
```

Reply:
Changed here: https://github.com/Mr-Smyth/task-manager/pull/17/commits/f2fd2326a3a1356127074befebe6c482ea0c274d